core/controllers/reader.py

```diff
@@ -81,6 +95,22 @@ def test_can_play(self, exploration_id, **kwargs):
 
 def _get_exploration_player_data(
         exploration_id, version, collection_id, can_edit):
+    """Get's exploration data which is returned as a dictionary.
+
+    Args:
+        exploration_id: str. The id of the exploration to get metadata.
+        version: int. The version of the exploration.
+        collection_id: str. The id of the collection which the exploration
+            belongs to.
+        can_edit: bool. If the exploration is editable by the learner.
+
+    Returns:
+        dict. A dictionary with metadata pertaining the exploration
+            such as  GADGET_SPECS,can_edit,title, version etc.
+
+    Raises:
+        Exception: The collection not in current available collections.
```

Comment:
There are multiple reasons for Exceptions here, and all of these should be documented.